capabilities/writetarget/messages.go

```diff
@@ -0,0 +1,259 @@
+//nolint:gosec,errcheck // disable G115, errcheck
+package writetarget
+
+import (
+	"encoding/hex"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	monitor "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/beholder"
+
+	wt "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/monitoring/pb/platform"
+)
+
+// messageBuilder is a helper component to build monitoring messages
+type messageBuilder struct {
+	ChainInfo monitor.ChainInfo
+	CapInfo   capabilities.CapabilityInfo
+}
+
+// NewMessageBuilder creates a new message builder
+func NewMessageBuilder(chainInfo monitor.ChainInfo, capInfo capabilities.CapabilityInfo) *messageBuilder {
+	return &messageBuilder{
+		ChainInfo: chainInfo,
+		CapInfo:   capInfo,
+	}
+}
+
+// reportInfo contains the report data for the request
+type reportInfo struct {
+	reportContext []byte
+	report        []byte
+	signersNum    uint32
+
+	// Decoded report fields
+	reportID uint16
+}
+
+// requestInfo contains the request data for the capability triggered
+type requestInfo struct {
+	tsStart int64
+
+	node      string
+	forwarder string
+	receiver  string
+
+	request                 capabilities.CapabilityRequest
+	reportInfo              *reportInfo
+	reportTransmissionState *TransmissionState
+}
+
+func (m *messageBuilder) buildWriteError(i *requestInfo, code uint32, summary, cause string) *wt.WriteError {
```

Comment:
nit: A pointer receiver and `*requestInfo` are not needed here, no data is being edited, just read and processed.